### PR TITLE
[DPE-6161] Document terminology decisions

### PR DESCRIPTION
**Pull request by izmalk**
Add the Terminology section to the CONTRIBUTING.md file.
Fix terminology in Readme files, CONTRIBUTING.md, and yaml specs.
FIx the AKS word triggering codespell linter.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -69,6 +69,36 @@ tox run -e integration   # integration tests
...
+| Apache Kafka cluster | | Charmed Apache Kafka cluster |
+
+The full form must be used at least once per page.
+The full form must be used at the first entry to the page’s headings, body of text, callouts, and graphics.
+For subsequent usage, the full form can be substituted by alternatives.
+
 ## Canonical Contributor Agreement
 
-Canonical welcomes contributions to the Charmed Kafka Kubernetes Operator. Please check out our [contributor agreement](https://ubuntu.com/legal/contributors) if you're interested in contributing to the solution.
+Canonical welcomes contributions to the Charmed Apache Kafka K8s Operator. Please check out our [contributor agreement](https://ubuntu.com/legal/contributors) if you're interested in contributing to the solution.
```
Comment (deusebio) [inline CONTRIBUTING.md:104]:
```suggestion
Canonical welcomes contributions to the Charmed Apache Kafka K8s. Please check out our [contributor agreement](https://ubuntu.com/legal/contributors) if you're interested in contributing to the solution.
```

File: README.md [hunk 2]
```diff
@@ -7,18 +7,18 @@
 
 ## Overview
 
-The Charmed Kafka K8s Operator delivers automated operations management [from Day 0 to Day 2](https://codilime.com/blog/day-0-day-1-day-2-the-software-lifecycle-in-the-cloud-age/) on the [Apache Kafka](https://kafka.apache.org) event streaming platform deployed on top of a [Kubernetes cluster](https://kubernetes.io/). It is an open source, end-to-end, production ready data platform on top of cloud native technologies.
+The Charmed Apache Kafka K8s Operator delivers automated operations management [from Day 0 to Day 2](https://codilime.com/blog/day-0-day-1-day-2-the-software-lifecycle-in-the-cloud-age/) on the [Apache Kafka](https://kafka.apache.org) event streaming platform deployed on top of a [Kubernetes cluster](https://kubernetes.io/). It is an open source, end-to-end, production ready data platform on top of cloud native technologies.
```
Comment (deusebio) [inline README.md:10]:
```suggestion
Charmed Apache Kafka K8s delivers automated operations management [from Day 0 to Day 2](https://codilime.com/blog/day-0-day-1-day-2-the-software-lifecycle-in-the-cloud-age/) on the [Apache Kafka](https://kafka.apache.org) event streaming platform deployed on top of a [Kubernetes cluster](https://kubernetes.io/). It is an open source, end-to-end, production ready data platform on top of cloud native technologies.
```

File: README.md [hunk 3]
```diff
@@ -7,18 +7,18 @@
...
 
 The Charmed Operator can be found on [Charmhub](https://charmhub.io/kafka-k8s) and it comes with features such as:
 - Fault-tolerance, replication, scalability and high-availability out-of-the-box.
 - SASL/SCRAM auth for Broker-Broker and Client-Broker authentication enabled by default.
 - Access control management supported with user-provided ACL lists.
 
-As currently Kafka requires a paired ZooKeeper deployment in production, this operator makes use of the [ZooKeeper K8s Operator](https://github.com/canonical/zookeeper-k8s-operator) for various essential functions.
+As currently Apache Kafka requires a paired Apache ZooKeeper deployment in production, this operator makes use of the [Apache ZooKeeper K8s Operator](https://github.com/canonical/zookeeper-k8s-operator) for various essential functions.
```
Comment (deusebio) [inline README.md:17]:
```suggestion
As currently Apache Kafka requires a paired Apache ZooKeeper deployment in production, this operator makes use of the [Charmed Apache ZooKeeper K8s](https://github.com/canonical/zookeeper-k8s-operator) for various essential functions.
```

File: README.md [hunk 4]
```diff
@@ -31,12 +31,12 @@ The charm can be deployed in much smaller environments if needed.
 
 ## Usage
 
-This section demonstrates basic usage of the Charmed Kafka K8s operator. 
-For more information on how to perform typical tasks, see the How to guides section of the [Charmed Kafka K8s documentation](https://canonical.com/data/docs/kafka/k8s).
+This section demonstrates basic usage of the Charmed Apache Kafka K8s operator.
```
Comment (deusebio) [inline README.md:34]:
```suggestion
This section demonstrates basic usage of Charmed Apache Kafka K8s. 
```

File: README.md [hunk 5]
```diff
@@ -81,15 +81,15 @@ Use the same action without a password parameter to randomly generate a password
 
 ### Storage support
 
-Currently, the Charmed Kafka K8s Operator makes use of a 10 GB storage mount, tied to a [Kubernetes PVC](https://kubernetes.io/docs/concepts/storage/persistent-volumes/).
+Currently, the Charmed Apache Kafka K8s Operator makes use of a 10 GB storage mount, tied to a [Kubernetes PVC](https://kubernetes.io/docs/concepts/storage/persistent-volumes/).
```
Comment (deusebio) [inline README.md:84]:
```suggestion
Currently, Charmed Apache Kafka K8s makes use of a 10 GB storage mount, tied to a [Kubernetes PVC](https://kubernetes.io/docs/concepts/storage/persistent-volumes/).
```

File: README.md [hunk 6]
```diff
@@ -185,15 +185,15 @@ juju remove-relation zookeeper-k8s tls-certificates-operator
 
 ## Monitoring
 
-The Charmed Kafka K8s Operator comes with several exporters by default. The metrics can be queried by accessing the following endpoints:
+The Charmed Apache Kafka K8s Operator comes with several exporters by default. The metrics can be queried by accessing the following endpoints:
```
Comment (deusebio) [inline README.md:188]:
```suggestion
The Charmed Apache Kafka K8s comes with several exporters by default. The metrics can be queried by accessing the following endpoints:
```

File: README.md [hunk 7]
```diff
@@ -226,9 +226,9 @@ After this is complete, Grafana will show a new dashboard: `Kafka JMX Metrics`.
 
 ## Security
 
-For an overview of security features of the Charmed Kafka K8s Operator, see the [Security page](https://canonical.com/data/docs/kafka/k8s/e-security) in the Explanation section of the documentation.
+For an overview of security features of the Charmed Apache Kafka K8s Operator, see the [Security page](https://canonical.com/data/docs/kafka/k8s/e-security) in the Explanation section of the documentation.
```
Comment (deusebio) [inline README.md:229]:
```suggestion
For an overview of security features of the Charmed Apache Kafka K8s, see the [Security page](https://canonical.com/data/docs/kafka/k8s/e-security) in the Explanation section of the documentation.
```

File: README.md [hunk 8]
```diff
@@ -226,9 +226,9 @@ After this is complete, Grafana will show a new dashboard: `Kafka JMX Metrics`.
...
 
-Security issues in the Charmed Kafka K8s Operator can be reported through [LaunchPad](https://wiki.ubuntu.com/DebuggingSecurity#How%20to%20File). Please do not file GitHub issues about security issues.
+Security issues in the Charmed Apache Kafka K8s Operator can be reported through [LaunchPad](https://wiki.ubuntu.com/DebuggingSecurity#How%20to%20File). Please do not file GitHub issues about security issues.
```
Comment (deusebio) [inline README.md:231]:
```suggestion
Security issues in the Charmed Apache Kafka K8s can be reported through [LaunchPad](https://wiki.ubuntu.com/DebuggingSecurity#How%20to%20File). Please do not file GitHub issues about security issues.
```

File: README.md [hunk 9]
```diff
@@ -237,4 +237,4 @@ Please see the [Juju SDK docs](https://juju.is/docs/sdk) for guidelines on enhan
 
 ## License
 
-The Charmed Kafka K8s Operator is free software, distributed under the Apache Software License, version 2.0. For more information, see the [LICENSE](https://github.com/canonical/kafka-k8s-operator/blob/main/LICENSE) file .
+The Charmed Apache Kafka K8s Operator is free software, distributed under the Apache Software License, version 2.0. For more information, see the [LICENSE](https://github.com/canonical/kafka-k8s-operator/blob/main/LICENSE) file .
```
Comment (deusebio) [inline README.md:240]:
```suggestion
The Charmed Apache Kafka K8s is free software, distributed under the Apache Software License, version 2.0. For more information, see the [LICENSE](https://github.com/canonical/kafka-k8s-operator/blob/main/LICENSE) file .
```

**Review by deusebio — APPROVED** [review]
Looks good! Usual nipticks to refer to the solution rather than the operator when sensible

File: README.md (hunk 9 above)
Comment (izmalk) [reply README.md:240]:
```suggestion
Charmed Apache Kafka K8s is a free software, distributed under the Apache Software License, version 2.0. For more information, see the [LICENSE](https://github.com/canonical/kafka-k8s-operator/blob/main/LICENSE) file.
```
Note: This update in wording changes the meaning quite a bit. Previously, we talked about the operator. Now we are talking about the whole solution, including Apache Kafka. It has the same Licence, so I guess it's OK. But you should be aware of the gravity of this change. @deusebio

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -69,6 +69,36 @@ tox run -e integration   # integration tests
...
+Apache®, [Apache Kafka, Kafka®](https://kafka.apache.org/), [Apache ZooKeeper, ZooKeeper™](https://zookeeper.apache.org/) and their respective logos are either registered trademarks or trademarks of the [Apache Software Foundation](https://www.apache.org/) in the United States and/or other countries.
+
+For documentation in this repository the following conventions are applied (see the table below).
+
+| Full form | Alternatives | Incorrect examples |
+| -------- | ------- | ------- |
+| Apache Kafka | | Kafka |
+| Charmed Apache Kafka K8s | | Charmed Kafka K8s, Charmed Kafka |
+| Kafka Connect | | Kafka connect |
+| Kafka Brokers | | Kafka brokers|
```
Comment (marcoppenheimer) [inline CONTRIBUTING.md:95]: **todo:** Remove L95. `Kafka brokers` is correct.

File: README.md [hunk 11]
```diff
@@ -237,4 +237,4 @@ Please see the [Juju SDK docs](https://juju.is/docs/sdk) for guidelines on enhan
 
 ## License
 
-The Charmed Kafka K8s Operator is free software, distributed under the Apache Software License, version 2.0. For more information, see the [LICENSE](https://github.com/canonical/kafka-k8s-operator/blob/main/LICENSE) file .
+Charmed Apache Kafka K8s is a free software, distributed under the Apache Software License, version 2.0. For more information, see the [LICENSE](https://github.com/canonical/kafka-k8s-operator/blob/main/LICENSE) file.
```
Comment (marcoppenheimer) [inline README.md:240]: **todo:** Remove the `a`. Software is 'non-countable', so it's correct to say `is free software`.

**Review by marcoppenheimer — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 10 above)
Comment (izmalk) [reply CONTRIBUTING.md:95]: Both options are viable as long as we are consistent in using only one of them. That's not for me to decide what option is better. Can you please discuss that with @grobbie and @deusebio ?

File: README.md (hunk 11 above)
Comment (izmalk) [reply README.md:240]:
```suggestion
Charmed Apache Kafka K8s is free software, distributed under the Apache Software License, version 2.0. For more information, see the [LICENSE](https://github.com/canonical/kafka-k8s-operator/blob/main/LICENSE) file.
```

File: CONTRIBUTING.md (hunk 10 above)
Comment (deusebio) [reply CONTRIBUTING.md:95]: I really don't have strong opinions on this, as long as we are consistent, as Vladimir is also suggesting. I would go in the following order:
1. @grobbie 
2. @izmalk 
If neither of you have any opinion, I'm happy with what @marcoppenheimer suggests as well.

Comment (izmalk) [reply CONTRIBUTING.md:95]: I've discussed this with @grobbie and we've settled with `Apache Kafka brokers`.

Comment (izmalk) [reply CONTRIBUTING.md:95]:
```suggestion
| Apache Kafka brokers | | Kafka brokers, Apache Kafka Brokers |
```

**Review by marcoppenheimer — APPROVED** [review]

**Pull request merged**